security/fma-superchainerc20.md

```diff
@@ -0,0 +1,147 @@
+    - Introducing the `OptimismSuperchainERC20Factory` predeploy.
+    - Introducing the `OptimismSuperchainERC20` implementation.
+    - Introducing the `BeaconContract` predeploy used by the `OptimismSuperchainERC20` BeaconProxies.
+
+Below are references for this project:
+
+- The new contracts and updates are documented in the following specs:
+    - [Predeploys specs](https://github.com/ethereum-optimism/specs/blob/main/specs/interop/predeploys.md).
+    - [Token standard specs](https://github.com/ethereum-optimism/specs/blob/main/specs/interop/token-bridging.md)
+
+## Failure Modes and Recovery Paths
+
+---
+
+### Incorrect or bypassing `_homeChain` definition
+
+- **Description:** During a `SuperchainERC20` deployment, `_homeChain` is defined as the chain where the initial supply is issued, generally to the owner. This is verified by matching with `block.chainId`. If this check is incorrectly handled or bypassed, new supply could be created on other chains during deployment, unexpectedly inflating the total supply.
+- **Risk Assessment**: Medium
+    - Potential impact: High. Users will face dilution and experience other side effects where the supply directly influences the system.
+    - Likelihood: Low. Setting a different `_homeChain` value or modifying the contract to relax this condition will result in generating a different address.
+- **Mitigation**: A code audit should ensure that the `_homeChain` check is properly implemented.
+- **Detection**: Monitor the initial token supply at every deployment to detect any unexpected supply creation.
+- **Recovery Path(s)**: Depends on the token. For the basic and non-ownable `SuperchainERC20`, token redeployment and user migration will be required. Other types of tokens may opt for different recovery paths.
+
+### Message unable to finalize in Cross-Chain Transfers
+
+- **Description:** During a cross-chain transaction, tokens are burned on the source chain via `sendERC20` and minted on the destination chain via `relayERC20` through `SuperchainERC20Bridge` and utilizing the `L2toL2CrossDomainMessenger`. However, if the destination chain doesn’t listen to messages from the origin chain, the message will never be included in `CrossL2Inbox` and therefore the tokens will be burned on the origin but not minted on the destination. This results in users’ funds being stuck or perceived as lost.
+- **Risk Assessment**: High
+    - Potential impact: High. Users may lose access to their funds, indefinitely.
+    - Likelihood: Medium. Currently there is no way to stop a cross-chain transaction whose destination doesn’t include the origin chain in their dependency set.
+- Mitigation: To be defined at the architecture level. In the meantime, user interfaces should prevent users from initiating a message that cannot be relayed.
+- **Detection**: Monitor cross-chain message queues and maintain an updated list of chain dependencies to identify potential incompatibilities proactively.
```

Comment:
What does "To be defined at the architecture level" mean? Similarly for the detection, what tooling are we using and how exactly would it detect incompatibilities?

These are just sample questions to get at some general guidance for mitigations and detections—we typically want to be as detailed as possible, so if this failure mode occurs and we perform a retro, we can look back at the FMA and clearly identify why our mitigations or detections were insufficient.

We should also link to specific mitigations, for example if we say "we have tests to mitigate this", or "we have a runbook for this recovery path", let's link to those artifacts. If those tests or runbooks are not yet written, you would add an action item to the FMA to write them and link them afterwards